docs/docs/20-concepts/02-mcp-gateway/03-obot-registry.md

```diff
@@ -0,0 +1,288 @@
+# Obot's Open Source Registry
+Each Obot instance connects to our open source registry by default. You can follow this guide to add your MCP server to [Obot’s registry](https://github.com/obot-platform/mcp-catalog). You’ll either submit **just a catalog entry**, or **a catalog entry + a repackaged image** (for stdio servers).
+
+---
+
+## Step 0 — Pick your path
+
+**Question:** What type of MCP server are you shipping?
+
+- **Remote / Streamable HTTP**
+    
+    You expose an HTTP MCP endpoint (hosted) **or** you already have a Docker image that serves the streamable HTTP MCP endpoint.
+    
+    → You only need [**mcp-catalog**](https://github.com/obot-platform/mcp-catalog).
+    
+- **Stdio**
+    
+    You have a npx/uvx stdio MCP server
+    
+    → You’ll touch **both** repos: first [**mcp-images**](https://github.com/obot-platform/mcp-images) (to containerize it), then [**mcp-catalog**](https://github.com/obot-platform/mcp-catalog).
+    
+
+---
+
+## Step 1 — Create your catalog entry
+
+Fork from: `https://github.com/obot-platform/mcp-catalog`
+
+Add a file: `your-server-name.yaml`
+
+### Base Template (required for all servers)
+
+```yaml
+name: Your Server Name
+description: |
+  One-line summary of what this server does.
+
+  *## Features*
+  - Key capability 1 
+  - Key capability 2 
+
+  *## What you'll need to connect*
+  - API key from https://example.com/api-keys
+  - Account with "read" scope enabled
+  - (Optional) Optional
+
+metadata:
+  categories: category-a, category-b
+
+icon: https://example.com/icon.png
+repoURL: https://github.com/your-org/your-mcp-repo
+
+*# Environment variables your server needs (optional)*
+env:
+  - key: API_KEY                     *# Actual env var name*
+    name: API Key                    *# Display name in UI*
+    required: true                   *# Is this mandatory?*
+    sensitive: true                  *# Masks value in UI (for credentials)*
+    description: Your API key from the developer dashboard
+
+*# Preview your server's tools (list top 20 if you have many)*
+toolPreview:
```

Comment:
Is there a way for them to generate this? Or do they have to manually type it all out?

Reply:
people can generate this using AI.

Reply:
added a `Tip` for the workflow I describe above.